python/lib/eeg.py

```diff
@@ -711,7 +711,7 @@ def fetch_and_insert_annotation_files(
             extension = 'tsv',
             suffix = 'annotations',
             all_ = True,
-            full_search = True,
+            full_search = False,
```

Comment:
raw: (no annotations)
```
>>> bids_layout.get_nearest(
... '/data/incoming/BRK-Face13-EEGNET-derivs-nocode/sub-BRK0001/ses-V01/eeg/sub-BRK0001_ses-V01_task-faceFO_acq-eeg_eeg.edf', 
... return_type = 'tuple',
... strict = False,
... extension = 'tsv',
... suffix = 'annotations',
... all_ = True,
... full_search = False,
... subject='BRK0001'
... )
[]
```

derivatives: (annotations)
```
>>> bids_layout.get_nearest(
... '/data/incoming/BRK-Face13-EEGNET-final/derivatives/EEG-IP-L/sub-BRK0001/ses-V01/eeg/sub-BRK0001_ses-V01_task-faceFO_desc-ll_desc-qcr_annotations.tsv', 
... return_type = 'tuple',
... strict = False,
... extension = 'tsv',
... suffix = 'annotations',
... all_ = True,
... full_search = False,
... subject='BRK0001'
... )
[<BIDSDataFile filename='/data/incoming/BRK-Face13-EEGNET-final/derivatives/EEG-IP-L/sub-BRK0001/ses-V01/eeg/sub-BRK0001_ses-V01_task-faceFO_desc-ll_desc-qcr_annotations.tsv'>]
>>> 
```

Reply:
mmm, in my case, nothing gets returned when full_search = False

```
bids_layout.get_nearest(
    '/Users/cmadjar/Data/LORIS_imaging_data/incoming/EEG/ds_bids_cbm_loris_24_11_21_small/sub-CBM00002/eeg/sub-CBM00002_task-protmap_eeg.edf',
    return_type='tuple',
    strict=False,
    extension='tsv',
    suffix='annotations',
    all_=True,
    full_search=False,
    subject='CBM00002'
)
[]
```

But if full search is set to True, then pybids can see the derivatives annotations:
```
>>> bids_layout.get_nearest(
...     '/Users/cmadjar/Data/LORIS_imaging_data/incoming/EEG/ds_bids_cbm_loris_24_11_21_small/sub-CBM00002/eeg/sub-CBM00002_task-protmap_eeg.edf',
...     return_type='tuple',
...     strict=False,
...     extension='tsv',
...     suffix='annotations',
...     all_=True,
...     full_search=True,
...     subject='CBM00002'
... )
[<BIDSDataFile filename='/Users/cmadjar/Data/LORIS_imaging_data/incoming/EEG/ds_bids_cbm_loris_24_11_21_small/derivatives/NeuronicEEG/sub-CBM00002/eeg/sub-CBM00002_task-protmap_annotations.tsv'>]
```

What is your version of pybids? Mine is `pybids, version 0.14.0` Maybe the behaviour is different depending on the version of pybids?

Reply:
@cmadjar You are passing the wrong .edf file to the query. The edf file passed in should be the one nested under the `/derivatives` folder since the annotation file is associated to this PhysioFile. That is why `Full_search=false` is not working for you.

Reply:
@jesscall there is no .edf files in the derivatives folder of my dataset. There is only *annotations.tsv and *annotations.json in the derivatives folder and it is the annotations that goes with the raw EDF. It does not look like an EEG file has to be present in the derivatives folder as well, especially when it is identical to the raw file.